src/mavedb/scripts/populate_mapped_hgvs.py

```diff
@@ -0,0 +1,188 @@
+                    if response.status_code != 200:
+                        logger.error(
+                            f"Failed for variant {variant_urn} to query ClinGen API for {mapped_variant.clingen_allele_id}: {response.status_code}"
+                        )
+                        continue
+                    data = response.json()
+                    if mapped_variant.clingen_allele_id.startswith("CA"):
+                        if data.get("genomicAlleles"):
+                            for allele in data["genomicAlleles"]:
+                                if allele.get("referenceGenome") == "GRCh38" and allele.get("hgvs"):
+                                    hgvs_g = allele["hgvs"][0]
+                                    break
+                        if target_is_coding:
+                            if data.get("transcriptAlleles"):
+                                if transcript_accession:
+                                    for allele in data["transcriptAlleles"]:
+                                        if allele.get("hgvs"):
+                                            for hgvs_string in allele["hgvs"]:
+                                                hgvs_reference_sequence = hgvs_string.split(":")[0]
+                                                if transcript_accession == hgvs_reference_sequence:
+                                                    hgvs_c = hgvs_string
+                                                    break
+                                        if hgvs_c:
+                                            if allele.get("proteinEffect"):
+                                                hgvs_p = allele["proteinEffect"].get("hgvs")
+                                            break
+                                else:
+                                    # no transcript specified, use mane if available
+                                    for allele in data["transcriptAlleles"]:
+                                        if allele.get("MANE"):
+                                            # TODO consider prioritizing mane select over other mane statuses
+                                            hgvs_c = allele["MANE"].get("nucleotide", {}).get("RefSeq", {}).get("hgvs")
+                                            hgvs_p = allele["MANE"].get("protein", {}).get("RefSeq", {}).get("hgvs")
+                                            break
+
+                    elif mapped_variant.clingen_allele_id.startswith("PA"):
+                        # if PA, assume that assay was performed at amino acid level, so only provide hgvs_p
+                        if data.get("aminoAcidAlleles"):
+                            for allele in data["aminoAcidAlleles"]:
+                                if allele.get("hgvs"):
+                                    hgvs_p = allele["hgvs"][0]
+                                    break
+
+                # TODO should we check that assay level hgvs mtches either g. or p.?
```

Comment:
Could be a good check if we want to fail the population if they don't match. If we are fine populating when they don't match, we can skip it.